course/frontend/array-shuffle/scripts/main.js

```diff
@@ -0,0 +1,36 @@
+// # Challenge ShuffleArray!
+
+/**
+ * Array.random() Crea un clon exacto de un array y desordena aleatoriamente sus elementos.
+ *
+ * @code
+ *  var originalArray = [1,2,3,'a','b','c'];
+ * 	var newArray = originalArray.random();
+ * @endcode
+ *
+ * @return {array} clonedArray. Retorna un nuevo array.
+ */
+Array.prototype.random = function (){
+	var clonedArray = [];
+	for (var i = 0; i < this.length; i++) {
+		clonedArray.push(this[i]);
+	}
+	// clonedArray == just a clone of "this" (the original array)!
+	var currentIndex = clonedArray.length;
+	var temporaryValue, randomIndex;
```

Comment:
NOTE temp- prefix may work well here, as it is well-known and accepted everywhere (i.e. tempValue, instead of temporaryValue (too long)).

Reply:
ok